### PR TITLE
Clarify monitor muting endpoints

### PR DESCRIPTION
**Pull request by wang-arthur**
- Adjust dogshell descriptions to distinguish between mute_all/unmute_all and mute/unmute
- Add new monitor unmute arg (`all_scopes`) to allow clearing all mute settings for a given monitor
- Update monitor mute tests
- Include additional information in 403 response exceptions

### CODE REVIEW

File: datadog/api/base.py [hunk 1]
```diff
@@ -124,6 +124,14 @@ def request(cls, method, path, body=None, attach_host_name=False, response_forma
...
                     pass
                 else:
+                    if e.response.text:
+                        try:
+                            error_msg = json.loads(e.response.text).get('errors')
+                            if error_msg:
+                                raise requests.exceptions.HTTPError("{0} ({1})".format(
+                                                                    e.message, error_msg[0]))
+                        except json.JSONDecodeError:
+                            pass
```
Comment (yannmh) [inline datadog/api/base.py]: Could you provide further details about the use case for this please ?

Comment (wang-arthur) [inline datadog/api/base.py]: Sure! Some 403 responses include custom error messages. For example, if a user tries to `$ dog monitor unmute_all` when the `mute_all` setting isn't enabled, the 403 response includes the error message: `'Monitoring is not currently globally muted'`. Simply raising `HttpError`, however, just gives the user a generic `'403 Client forbidden error'`.

File: tests/integration/api/test_api.py [hunk 2]
```diff
@@ -652,7 +653,7 @@ def test_monitor_muting(self):
         eq(dt, None)  # No response is expected.
 
         # We shouldn't be able to mute a simple alert on a scope.
-        assert_raises(ApiError, dog.Monitor.mute, monitor_id, scope='env:staging')
+        assert_raises(HTTPError, dog.Monitor.mute, monitor_id, scope='env:staging')
```
Comment (yannmh) [inline tests/integration/api/test_api.py]: The exception is proper to the API behavior so I think an ApiError makes more sense in this case.

**Comment by yannmh** [comment]
Thanks alot  @wang-arthur. It looks great :+1: 

Before merging it, I think we should clarify which behaviors fall under the `HTTPError` or`ApiError` exceptions.

File: tests/integration/api/test_api.py (hunk 2 above)
Comment (yannmh) [inline tests/integration/api/test_api.py]: Let's switch to an ApiError instead @wang-arthur.

File: datadog/api/base.py [hunk 3]
```diff
@@ -124,6 +124,14 @@ def request(cls, method, path, body=None, attach_host_name=False, response_forma
                 if e.response.status_code == 404 or e.response.status_code == 400:
                     pass
```
Comment (yannmh) [inline datadog/api/base.py]: 403 should probably be caught in this block and raise at this point https://github.com/DataDog/datadogpy/blob/clarify_mute_all/datadog/api/base.py#L160

**Comment by wang-arthur** [comment]
Thanks @yannmh! I made the revisions we discussed and cleaned up the commit history.

**Comment by yannmh** [comment]
Thanks a lot @wang-arthur. Merging it for the v0.7.0 release :ship:

**Pull request merged**